app/views/admin/roles/_form.html.erb

```diff
@@ -5,48 +5,116 @@
 
 <%= form_for role, as: :role, url: role_url_for(role) do |form| %>
   <%= form.errors %>
+    <%= render "govuk_publishing_components/components/input", {
+        label: {
+          text: "Role title"
+        },
+        value: role.name,
+        name: "role[name]",
+        id: "role_name",
+        type: "text"}
+    %>
 
-  <fieldset>
-    <%= form.text_field :name, label_text: 'Role title' %>
+    <%= render "govuk_publishing_components/components/select", {
+      id: "role_type_id",
+      label: "Role type",
+      name: "role[role_type]",
+      options:
+        [{value: nil, text: nil}] + role_type_options.map do |group, name_vs_type|
```

Comment:
sorry i know this is a pain since we don't use an erb linter, but if we did it would tell us to leave a space on the outside of curly braces. So

```
[ {value: nil, text: nil} ] + role_type_options.map do |group, name_vs_type|
```

and 

```
{text: name, value: type, selected: role.role_type == type }

```

Generally we also move multi key/value hashes over multiple lines for readability.

So: 

```
name_vs_type.map do |name, type|
  { 
     text: name, 
     value: type, 
      selected: role.role_type == type 
  }
end
```

Reply:
Yep, should have tidied these instances up. ✅